Sources/Commands/Workspace.swift

```diff
@@ -521,6 +527,34 @@ public class Workspace {
         return try PackageGraphLoader().load(rootManifest: rootManifest, externalManifests: externalManifests)
     }
 
+    /// Removes the clone and checkout of the provided specifier.
+    func remove(specifier: RepositorySpecifier) throws {
+        guard let dependency = dependencyMap[specifier] else {
+            fatalError("This should never happen, trying to remove \(specifier) which isn't in workspace")
+        }
+
+        // Inform the delegate.
+        delegate.removing(repository: dependency.repository.url)
+
+        // Remove the repository from dependencies.
+        dependencyMap[dependency.repository] = nil
+
+        // Remove the checkout.
+        let dependencyPath = checkoutsPath.appending(dependency.subpath)
+        let checkedOutRepo = try checkoutManager.provider.openCheckout(at: dependencyPath)
+        guard !checkedOutRepo.hasUncommitedChanges() else {
+            throw WorkspaceOperationError.hasUncommitedChanges(repo: dependencyPath)
+        }
+        try removeFileTree(dependencyPath)
+
+        // Remove the clone.
+        let handle = checkoutManager.lookup(repository: dependency.repository)
```

Comment:
Would it make more sense for `checkoutManager` to expose a `remove(repository:)`? It feels add to do a lookup just to remove. I'm fine landing that change after this...